### PR TITLE
Allow trust env parameter to be defined in extra options of HTTP Connection

### PR DESCRIPTION
**Pull request by dabla**
Small PR which also allows trust env to be defined in the extra options of the Connection
---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/providers/http/hooks/http.py [hunk 1]
```diff
@@ -436,6 +438,8 @@ def _process_extra_options_from_connection(cls, conn: Connection, extra_options:
             extra_options["allow_redirects"] = allow_redirects
         if max_redirects is not None and "max_redirects" not in extra_options:
             extra_options["max_redirects"] = max_redirects
+        if trust_env is not None and "trust_env" not in extra_options:
```
Comment (dirrao) [inline airflow/providers/http/hooks/http.py:441]:
```suggestion
        if trust_env and "trust_env" not in extra_options:
```

**Review by dirrao — CHANGES_REQUESTED** [review]
Can you let us know the use case of this additional parameter?

File: airflow/providers/http/hooks/http.py (hunk 1 above)
Comment (dabla) [reply airflow/providers/http/hooks/http.py:441]: I would not do that as trust_env is a boolean, and if not None but False and not already defined in the extra_option, we would then define it in the extra_options.  In your proposition we would only set if True, but that's also already the default value, we just want to use this parameter to be able to set it to False

**Comment by dabla** [comment]
> Can you let us know the use case of this additional parameter?

Yes this is when you want the requests library to ignore the environment variables.  In our case we have default proxy environment variables defined, but in some cases we need to pass through another proxy, so even if we specify the proxy parameter, the request library will still use the one from the env variable unless we tell it to ignore it.

File: airflow/providers/http/hooks/http.py (hunk 1 above)
Comment (dabla) [reply airflow/providers/http/hooks/http.py:441]: Will look to try to test this particular situation

Comment (dabla) [reply airflow/providers/http/hooks/http.py:441]: Assert added when case of trust_env False

test_async_request_uses_connection_extra_with_requests_parameters

**Review by potiuk — APPROVED** [review]
LGTM

File: airflow/providers/http/hooks/http.py (hunk 1 above)
Comment (potiuk) [reply airflow/providers/http/hooks/http.py:441]: @dirrao -> does it solve your question?

**Comment by dabla** [comment]
Is it possible to merge this PR as this also impacts this [PR](https://github.com/apache/airflow/pull/35591/) so I can fix the merge conflicts on the later one and continue working on it? Thanks :)

**Pull request merged**